src/ipa/rpi/controller/rpi/sync.cpp

```diff
@@ -119,100 +144,198 @@ void Sync::initialise()
 	socket_ = -1;
 }
 
+/* No idea what this does*/
 void Sync::switchMode([[maybe_unused]] CameraMode const &cameraMode, [[maybe_unused]] Metadata *metadata)
 {
 	syncReady_ = false;
 	frameCount_ = 0;
 	readyCountdown_ = 0;
 }
 
+
+/* Most important part, algorithm*/
 void Sync::process([[maybe_unused]] StatisticsPtr &stats, Metadata *imageMetadata)
 {
 	SyncPayload payload;
-	SyncParams local {};
-	SyncStatus status {};
+	SyncParams local{};
+	SyncStatus status{};
 	imageMetadata->get("sync.params", local);
+	static int64_t lag = 0;
+	static bool initialise = false;
+
 
 	if (!frameDuration_) {
 		LOG(RPiSync, Error) << "Sync frame duration not set!";
 		return;
 	}
+	if (mode_ == Mode::Off) {
+		return;
+	}
 
 	if (mode_ == Mode::Server) {
+
+		if (!initialise){
+			initialise = true;
+			Sync::initialise_socket();
+		}
+
+		static ClockRecovery trendingClock(local.wallClock, local.sensorTimestamp, syncPeriod_, lineFitting_);
+		
+		static int64_t lastWallClock = local.wallClock;
+
+		int mul = (local.wallClock - lastWallClock - frameDuration_.get<std::micro>() * 0.5) / frameDuration_.get<std::micro>();
```

Comment:
`mul` feels like a slightly strange name. Can we think of something better? `frameDiff`?